packages/cli/src/workflows/workflows.controller.ts

```diff
@@ -396,6 +396,44 @@ export class WorkflowsController {
 		return true;
 	}
 
+	@Post('/:workflowId/archive')
+	@ProjectScope('workflow:delete')
+	async archive(req: WorkflowRequest.Delete) {
+		const { workflowId } = req.params;
+
+		const workflow = await this.workflowService.archive(req.user, workflowId);
+		if (!workflow) {
+			this.logger.warn('User attempted to archive a workflow without permissions', {
+				workflowId,
+				userId: req.user.id,
+			});
+			throw new ForbiddenError(
+				'Could not archive the workflow - you can only archive workflows owned by you',
+			);
+		}
+
+		return workflow;
```

Comment:
Should we return the workflow or simply a success response? Same below.

Reply:
On archive/unarchive the `versionId` changes and the new `versionId` is needed on the editor, else the user can see errors related to update conflicts when changing workflow from archived to unarchived, which is why I made this return the workflow.